test_scripts/Defects/8_2/3888/common.lua

```diff
@@ -0,0 +1,314 @@
+local color = require("user_modules/consts").color
+local SDL = require('SDL')
+local runner = require('user_modules/script_runner')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Module ]]
+local m = { }
+
+--[[ Proxy Functions ]]
+m.Title = runner.Title
+m.Step = runner.Step
+m.preconditions = actions.preconditions
+m.postconditions = actions.postconditions
+m.start = actions.start
+m.registerAppWOPTU = actions.registerAppWOPTU
+m.activateApp = actions.activateApp
+m.spairs = utils.spairs
+m.connectMobile = actions.mobile.connect
+
+--[[ Local Variables ]]
+local addSubMenuParams = {
+  menuID = 1000,
+  position = 500,
+  menuName ="Sub Menu"
+}
+
+--[[ Common Variables ]]
+m.testCases = {
+  with_vr_command = {
+    parameters = {
+      interactionChoiceSetID = 1001,
+      choiceSet = {
+        {
+          choiceID = 1001,
+          menuName ="Choice1001",
+          vrCommands = {
+            "Choice1001"
+          }
+        }
+      }
+    },
+    vrCommandTimes = 1,
+    vrAddCommandRequest = {
+      cmdID = 1001,
+      type = "Choice",
+      vrCommands = {
+        "Choice1001"
+      }
+    },
+    performInteractionVR = {
+      choiceId = 1001,
+      result = "SUCCESS"
+    },
+    performInteractionMANUAL = {
+      choiceId = 1001,
+      result = "SUCCESS"
+    }
+  },
+  without_vr_command = {
+    parameters = {
+      interactionChoiceSetID = 1002,
+      choiceSet = {
+        {
+          choiceID = 1002,
+          menuName ="Choice1002"
+        }
+      }
+    },
+    vrCommandTimes = 0,
+    performInteractionVR = {
+      choiceId = 1002,
+      result = "INVALID_DATA"
+    },
+    performInteractionMANUAL = {
+      choiceId = 1002,
+      result = "SUCCESS"
+    }
+  }
+}
+m.hashID = 0
+m.resumeFailedResult = "RESUME_FAILED"
+
+--[[ Common Function ]]
+function m.createInteractionChoiceSet(pParams)
+  local cid = actions.getMobileSession():SendRPC("CreateInteractionChoiceSet", pParams.parameters)
+  actions.getHMIConnection():ExpectRequest("VR.AddCommand", pParams.vrAddCommandRequest or {})
+  :Do(function(_, data)
+      actions.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+  :ValidIf(function(_, data)
+      if data.params.grammarID ~= nil then
+        pParams.vrAddCommandRequest.grammarID = data.params.grammarID
+        return true
+      else
+        return false, "grammarID should not be empty"
+      end
+    end)
+  :Times(pParams.vrCommandTimes)
+  actions.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+  actions.getMobileSession():ExpectNotification("OnHashChange")
+  :Do(function(_, data)
+      m.hashID = data.payload.hashID
+    end)
+end
+
+function m.unexpectedDisconnect()
+  actions.mobile.disconnect()
+  actions.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered",
+    { appID = actions.getHMIAppId(), unexpectedDisconnect = true })
```

Comment:
Is this a standard function anywhere?

Reply:
Seems that the function is defined differently in various common files so there's no base implementation to use. It is defined  in `commonSmoke.lua` but the commonSmoke implementation deletes the mobileSession while this implementation only does a disconnect